app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -229,93 +194,35 @@ export default Component.extend(KeyEnterEscape, {
     const { isIOS, isAndroid, isOpera } = this.capabilities;
     const showAtEnd = isMobileDevice || isIOS || isAndroid || isOpera;
 
-    const boundaryPosition = this._getRangeBoundaryRect(firstRange, showAtEnd);
+    if (showAtEnd) {
+      this.placement = "bottom-start";
+    }
 
     // change the position of the button
     schedule("afterRender", () => {
       if (!this.element || this.isDestroying || this.isDestroyed) {
         return;
       }
 
-      let top = 0;
-      let left = 0;
-      const pxFromSelection = 5;
-
-      if (showAtEnd) {
-        // The selection-handles on iOS have a hit area of ~50px radius
-        // so we need to make sure our buttons are outside that radius
-        // Apply the same logic on all mobile devices for consistency
-
-        top = boundaryPosition.bottom + pxFromSelection;
-        left = boundaryPosition.left;
-
-        const safeRadius = 50;
+      this.textRange = new VirtualElementFromTextRange(".cooked");
```

Comment:
This should be initialized (like `placement: "top-start"` above). 

And then in `willDestroyElement` it can be reset and the event listeners can be cleaned up via `this.textRange?.clearListeners()`.

Reply:
Yes, IMO, setup/clear listeners shouldn't be in the text range function but in the quote-button. The text range should only be responsible to create a virtual element of the right dimension/position

Reply:
> This should be initialized (like placement: "top-start" above).

This seemed to be causing issues since there is no text selection present initially, but the tests are now passing anyway after moving the setup/clear of listeners to `quote-button.js`. So I think this is resolved.